aten/src/ATen/FunctionalInverses.cpp

```diff
@@ -95,252 +95,312 @@ static Tensor unsqueeze_copy_to(const Tensor & self, IntArrayRef dim, c10::SymIn
     return Tensor();
 }
 
-Tensor FunctionalInverses::view_as_real_copy_inverse(const Tensor& base, const Tensor& mutated_view, bool reapply_views) {
-    if (reapply_views) {
+Tensor FunctionalInverses::view_as_real_copy_inverse(const Tensor& base, const Tensor& mutated_view, InverseReturnMode inverse_return_mode) {
+    if (inverse_return_mode != InverseReturnMode::NeverView) {
       return at::view_as_complex(mutated_view);
     } else {
       return at::view_as_complex_copy(mutated_view);
     }
 }
 
-Tensor FunctionalInverses::view_as_complex_copy_inverse(const Tensor& base, const Tensor& mutated_view, bool reapply_views) {
-    if (reapply_views) {
+Tensor FunctionalInverses::view_as_complex_copy_inverse(const Tensor& base, const Tensor& mutated_view, InverseReturnMode inverse_return_mode) {
+    if (inverse_return_mode != InverseReturnMode::NeverView) {
       return at::view_as_real(mutated_view.resolve_conj());
     } else {
       return at::view_as_real_copy(mutated_view.resolve_conj());
     }
 }
 
-Tensor FunctionalInverses::_conj_copy_inverse(const Tensor& base, const Tensor& mutated_view, bool reapply_views) {
-    if (reapply_views) {
+Tensor FunctionalInverses::_conj_copy_inverse(const Tensor& base, const Tensor& mutated_view, InverseReturnMode inverse_return_mode) {
+    if (inverse_return_mode != InverseReturnMode::NeverView) {
       return at::_conj(mutated_view);
     } else {
       return at::_conj_copy(mutated_view);
     }
 }
 
-Tensor FunctionalInverses::_neg_view_copy_inverse(const Tensor& base, const Tensor& mutated_view, bool reapply_views) {
-    if (reapply_views) {
+Tensor FunctionalInverses::_neg_view_copy_inverse(const Tensor& base, const Tensor& mutated_view, InverseReturnMode inverse_return_mode) {
+    if (inverse_return_mode != InverseReturnMode::NeverView) {
       return at::_neg_view(mutated_view);
     } else {
       return at::_neg_view_copy(mutated_view);
     }
 }
 
-Tensor FunctionalInverses::as_strided_copy_inverse(const Tensor& base, const Tensor& mutated_view, bool reapply_views, at::SymIntArrayRef size, at::SymIntArrayRef stride, c10::optional<c10::SymInt> storage_offset) {
-    // Pessimism: we can't reapply views for as_strided_scatter.
-    return base.as_strided_scatter_symint(mutated_view, size, stride, std::move(storage_offset));
+Tensor FunctionalInverses::as_strided_copy_inverse(const Tensor& base, const Tensor& mutated_view, InverseReturnMode inverse_return_mode, at::SymIntArrayRef size, at::SymIntArrayRef stride, c10::optional<c10::SymInt> storage_offset) {
+    if (inverse_return_mode == InverseReturnMode::AlwaysView) {
+      // NB: assumes mutated_view is a narrowed view of base.
+      // We should NOT do this for functionalization
+      return mutated_view.as_strided_symint(
+          base.sym_sizes(), base.sym_strides(), base.sym_storage_offset());
+    } else {
+      return base.as_strided_scatter_symint(mutated_view, size, stride, std::move(storage_offset));
```

Comment:
lol - I wonder if `as_strided_scatter` is any easier for a backend/subclass to implement than as_strided.

(this op will only ever be hit if the user code included an as_strided, and at that point the backend/subclass will need to have implemented as_strided anyway. But whatever, the consistency here is probably fine).

Reply:
Haha good point! I do still need the view version, and I feel better not changing the existing behavior if I can help it, even it it's not likely to impact existing subclasses. That + the consistency you mentioned